### ISSUE TITLE
Is there any plan of developing softmax-weight function for using 12 hidden BERT layer? 

### ISSUE THREAD

**Issue by izuna385** [huggingface/pytorch-transformers#814]
Thanks for developing very nice/useful library.

My question is about using 12/(or in large model, more) hidden layer.
First, Does how to use hidden layers depend on the downstream task?
(Say, concat, average, only final layer, only mean of top 4 layer, etc...)

For using all layer, I think it's good to use softmax weight. During training , hidden layer's feature is fix but weight is learned for the task. So second question is, Is there any plan of developing softmax-weight function for using 12 hidden BERT layer? 

Thanks

**Comment by thomwolf** [comment]
Yes we might add a module for scalar mixture of layers like the one of AllenNLP, for instance (https://github.com/allenai/allennlp/blob/master/allennlp/modules/scalar_mix.py).

**Comment by izuna385** [comment]
I'm looking forward to see that in also pytorch-transformer.
Again, thanks. I'll keep track on this repository.
